### PR TITLE
[netif-posix] add an option to persistent interface

### PR DESCRIPTION
**Pull request by kghost**
In order to fix a matter [issue](https://github.com/project-chip/connectedhomeip/issues/21881), we need the feature to persistent the interface when restarting ot-br

This PR adds a command line option to persistent interface.

### CODE REVIEW

File: src/posix/main.c [hunk 1]
```diff
@@ -141,6 +141,7 @@ enum
     OT_POSIX_OPT_DRY_RUN                 = 'n',
     OT_POSIX_OPT_HELP                    = 'h',
     OT_POSIX_OPT_INTERFACE_NAME          = 'I',
+    OT_POSIX_OPT_PERSISTENT_INTERFACE    = 'p',
```
Comment (bukepo) [inline src/posix/main.c:144]: If persistent is better in most cases, I don't think we need add an option for not enabling it.

Comment (simonlingoogle) [reply src/posix/main.c:144]: Maybe it's still a good idea to add an option. 
Some users may still prefer temporary interface?

File: src/posix/platform/netif.cpp [hunk 2]
```diff
@@ -1598,41 +1602,62 @@ static void processMLDEvent(otInstance *aInstance)
...
     memset(&ifr, 0, sizeof(ifr));
-    ifr.ifr_flags = IFF_TUN | IFF_NO_PI | static_cast<short>(IFF_TUN_EXCL);
+    ifr.ifr_flags = IFF_TUN | IFF_NO_PI;
+    if (!aPlatformConfig->mPersistentInterface)
+    {
+        ifr.ifr_flags |= static_cast<short>(IFF_TUN_EXCL);
+    }
 
-    if (aInterfaceName)
+    interfaceName = aPlatformConfig->mInterfaceName;
```
Comment (simonlingoogle) [inline src/posix/platform/netif.cpp:1620]: What happens if `--persistent-interface` is specified but interface name is not given?
Is it a meaningful usecase?

File: src/posix/platform/netif.cpp [hunk 3]
```diff
@@ -1598,41 +1602,62 @@ static void processMLDEvent(otInstance *aInstance)
...
+    }
+
     VerifyOrDie(ioctl(sTunFd, TUNSETLINK, ARPHRD_VOID) == 0, OT_EXIT_ERROR_ERRNO);
 
-    strncpy(deviceName, ifr.ifr_name, deviceNameLen);
+    if (aPlatformConfig->mPersistentInterface)
+    {
+        if (gInstance != nullptr && otIp6IsEnabled(gInstance))
+        {
+            SetLinkState(gInstance, true);
```
Comment (simonlingoogle) [inline src/posix/platform/netif.cpp:1648]: Turning the link on may not be necessary?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with nits

File: src/posix/platform/netif.cpp (hunk 2 above)
Comment (kghost) [reply src/posix/platform/netif.cpp:1620]: In this case, it work as same as before, a device with name "wpanX" will be created, then we make it persistent.

File: src/posix/platform/netif.cpp (hunk 3 above)
Comment (kghost) [reply src/posix/platform/netif.cpp:1648]: Done

File: src/posix/main.c (hunk 1 above)
Comment (kghost) [reply src/posix/main.c:144]: Agree, This is not fully compatible with previous behavior.

File: src/posix/platform/netif.cpp (hunk 2 above)
Comment (simonlingoogle) [reply src/posix/platform/netif.cpp:1620]: What happens if we run `-I wpan0 --persistent-interface` first, and then run `-I wpan0`?

Can the second run be successfully, and the wpan0 interface be removed when the process quit?

Comment (kghost) [reply src/posix/platform/netif.cpp:1620]: No, the second run fails, due to not able to create `wpan0`

Comment (simonlingoogle) [reply src/posix/platform/netif.cpp:1620]: Should we make sure that the second run can be successful?

For example, we may enable `--persistent-interface` on one product, and later decide to disable `--persistent-interface`. Current implementation would not allow simply removing `--persistent-interface` to work.

Comment (kghost) [reply src/posix/platform/netif.cpp:1620]: I requires a reboot to update the firmware, right ? after a reboot, the device will be removed.

File: src/posix/platform/netif.cpp [hunk 4]
```diff
@@ -1598,41 +1604,54 @@ static void processMLDEvent(otInstance *aInstance)
...
     VerifyOrDie(ioctl(sTunFd, TUNSETIFF, static_cast<void *>(&ifr)) == 0, OT_EXIT_ERROR_ERRNO);
-    VerifyOrDie(ioctl(sTunFd, TUNSETLINK, ARPHRD_VOID) == 0, OT_EXIT_ERROR_ERRNO);
 
-    strncpy(deviceName, ifr.ifr_name, deviceNameLen);
+    strncpy(gNetifName, ifr.ifr_name, sizeof(gNetifName));
+
+    if (aPlatformConfig->mPersistentInterface)
+    {
+        VerifyOrDie(ioctl(sTunFd, TUNSETPERSIST, 1) == 0, OT_EXIT_ERROR_ERRNO);
+        SetLinkState(gInstance, false); // Set link down to reset the tun configuration.
```
Comment (bukepo) [inline src/posix/platform/netif.cpp:1641]: Could you add comments about what will be reset?

Comment (kghost) [reply src/posix/platform/netif.cpp:1641]: Add more comments

**Review by bukepo — APPROVED** [review]

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**